lib/auth/server_info.go

```diff
@@ -88,6 +88,13 @@ func getServerInfoNames(node types.Server) []string {
 
 func (a *Server) setLabelsOnNodes(ctx context.Context, nodes []types.Server) (failedUpdates int, err error) {
 	for _, node := range nodes {
+		// EICE Node labels can't be updated using the Inventory Control Stream because there's no reverse tunnel.
+		// Labels are updated by the DiscoveryService during 'Server.handleEC2Instances'.
+		// The same is valid for OpenSSH Nodes.
+		if node.IsOpenSSHNode() {
+			continue
+		}
```

Comment:
It didn't because there's no match
We look at the `spec.aws` metadata of the node and not at the labels.
Only labels are set when creating the resource.

https://github.com/gravitational/teleport/pull/35504#discussion_r1486102450

So we didn't had an error
But even after fixing the above and lookup the labels instead of the `spec.aws` metadata, we would not get the labels updated because we can't update the labels of a `openssh` node (there's no inventory control channel we can use).